components/brave_wallet/common/hex_utils.cc

```diff
@@ -130,7 +130,12 @@ bool HexValueToInt256(const std::string& hex_input, int256_t* out) {
   // This is the same as ~val + 1
   // To convert a positive number into a negative number, using the two’s
   // complement representation, invert all of the bits of the number + 1
+  if (val >> 255) {
```

Comment:
seems like we should just `CHECK` here if the actual value is larger than 255 bit? I think we want to attempt to treat the weird boost int256 the same as the BigInt in terms of max values

Reply:
or return false I guess would be fine and easier to test

Reply:
we already have a overflow case for it and it doesn't need extra change to pass